src/drive/ducks/files/FileOpenerExternal.js

```diff
@@ -0,0 +1,86 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import { connect } from 'react-redux'
+import { flowRight as compose } from 'lodash'
+
+import { Spinner, translate } from 'cozy-ui/react'
+import styles from './styles'
+import Viewer from 'viewer'
+import { fetchDocument, getDocument } from 'cozy-client'
+
+const doNothing = () => {}
+
+class FileOpener extends Component {
+  state = { loading: true }
+  componentWillMount() {
+    this.loadFileInfo()
+  }
+
+  async loadFileInfo() {
+    const { alert } = this.props
+    try {
+      await this.props.fetchFile()
+      // Go to the parent folder, we replace since we do not want
+      // to add a new history entry
+      // router.replace(`/folder/${fileInfo.attributes.dir_id}/file/${fileId}`)
+    } catch (e) {
+      console.warn(e)
+      // Go to the root folder, we replace since we do not want
+      // to add a new history entry
+      alert({
+        message: 'alert.could_not_open_file'
+      })
```

Comment:
What happens if the file does not exist ? I am not sure that the Alerter is mounted in the service, so maybe it will not work.. Maybe the simplest thing would be to ditch the `alert` mechanism and render an error message ?

Reply:
Actually it seems we never enter in the `catch` block. So I don't really know how to catch the error that occurs.